ui/src/components/FormModifications/FormModifications.test.tsx

```diff
@@ -0,0 +1,238 @@
+
+    return [componentParentElement, componentInput];
+};
+
+it('render fields with modifications correctly', async () => {
+    setUpConfigWithDefaultValue();
+    renderModalWithProps(props);
+
+    const verifyCorretDisplayForTextField = (field: string) => {
+        const [componentParentElement, componentInput] = getHTMLTextComponentsForField(field);
+        expect(componentParentElement).toBeInTheDocument();
+        expect(componentInput).toBeInTheDocument();
+    };
+
+    const verifyCorretDisplayForcheckboxField = (field: string) => {
+        const [componentParentElement, componentInput] = getHTMLChecboxComponentsForField(field);
+        expect(componentParentElement).toBeInTheDocument();
+        expect(componentInput).toBeInTheDocument();
+    };
+    verifyCorretDisplayForTextField(firstStandardTextField.field);
+    verifyCorretDisplayForTextField(secondStandardTextField.field);
+    verifyCorretDisplayForTextField(firstModificationField.field);
+    verifyCorretDisplayForTextField(secondModificationField.field);
+    verifyCorretDisplayForcheckboxField(thirdModificationField.field);
+});
+
+it('verify modification after text components change', async () => {
+    setUpConfigWithDefaultValue();
+    renderModalWithProps(props);
+    const firstValueToInput = 'a';
+    const secondValueToInput = 'aa';
+
+    const [componentParentElement, componentInput] = getHTMLTextComponentsForField(
+        firstStandardTextField.field
+    );
+    const [component2ParentElement, component2Input] = getHTMLTextComponentsForField(
+        secondStandardTextField.field
+    );
+
+    const componentMakingModsTextBox1 = document.querySelector(
+        `.${firstModificationField.field} input`
+    );
+
+    const mods1Field1 = findMods(
+        firstModificationField,
+        firstValueToInput,
+        firstStandardTextField.field
+    );
+    const mods1Field2 = findMods(
+        firstModificationField,
+        firstValueToInput,
+        secondStandardTextField.field
+    );
+
+    const mods2Field1 = findMods(
+        firstModificationField,
+        secondValueToInput,
+        firstStandardTextField.field
+    );
+    const mods2Field2 = findMods(
+        firstModificationField,
+        secondValueToInput,
+        secondStandardTextField.field
+    );
+
+    if (componentMakingModsTextBox1) {
+        await userEvent.type(componentMakingModsTextBox1, firstValueToInput);
+    }
+
+    const verifyAllProps = (
+        parentElement: Element | null | undefined,
+        input: Element | null | undefined,
+        mods?: { value?: string | number | boolean; help?: string; label?: string } | null
+    ) => {
+        expect(input?.getAttribute('value')).toEqual(mods?.value);
+        expect(parentElement?.textContent?.includes(mods?.help || 'fail')).toEqual(true);
+        expect(parentElement?.textContent?.includes(mods?.label || 'fail')).toEqual(true);
```

Comment:
this magic string `fail` puzzled me a lot. I realized you try to workaround TS error that caused by the fact you should not call `expect` conditionally.
Well. I would just suggest indeed fail if there is no value. How? by checking the value and throwing an error. Check out my little helper `invariant`

Amount of optional chaining is overwhelming. I would encourage to reduce it

Reply:
true, added invariant